public/app/features/alerting/unified/components/rules/CloudRules.tsx

```diff
@@ -48,24 +50,39 @@ export const CloudRules = ({ namespaces, expandAll }: Props) => {
     DEFAULT_PER_PAGE_PAGINATION
   );
 
+  const [createRuleSupported, createRuleAllowed] = useAlertingAbility(AlertingAction.CreateAlertRule);
+  const [viewExternalRuleSupported, viewExternalRuleAllowed] = useAlertingAbility(AlertingAction.ViewExternalAlertRule);
+
+  const canViewCloudRules = viewExternalRuleSupported && viewExternalRuleAllowed;
+
+  const canCreateGrafanaRules = createRuleSupported && createRuleAllowed;
+  const canMigrateToGMA = canCreateGrafanaRules && canViewCloudRules && config.featureToggles.alertingMigrationUI;
```

Comment:
Should this also check that there are datasources configured that the user can migrate? e.g. include `hasDataSourcesConfigured`